tests/unit/hive_metastore/test_workflows.py

```diff
@@ -70,3 +82,73 @@ def test_refresh_migration_status_is_refreshed(run_workflow, workflow):
     assert "DELETE FROM hive_metastore.ucx.migration_status" in ctx.sql_backend.queries
     assert "SHOW DATABASES" in ctx.sql_backend.queries
     # No "SHOW TABLE FROM" query as table are not mocked
+
+def test_refresh_migration_status_published_remained_tables(run_workflow):
+    # class LogCaptureHandler(logging.Handler):
+    #     def __init__(self):
+    #         super().__init__()
+    #         self.records = []
+    #
+    #     def emit(self, record):
+    #         self.records.append(record)
+
+
+    # # Setup custom log handler
+    # log_capture_handler = LogCaptureHandler()
+    # logger = logging.getLogger(__name__)
+    # logger.addHandler(log_capture_handler)
+    # logger.setLevel(logging.INFO)
+
+    # Setup mocks
+    custom_table_crawler = create_autospec(TablesCrawler)
+    custom_table_crawler.snapshot.return_value = [
+        Table(
+            object_type="EXTERNAL",
+            table_format="DELTA",
+            catalog="hive_metastore",
+            database="schema1",
+            name="table1",
+            location="s3://some_location/table1",
+            upgraded_to="ucx_default.db1_dst.dst_table1",
+        ),
+        Table(
+            object_type="EXTERNAL",
+            table_format="DELTA",
+            catalog="hive_metastore",
+            database="schema1",
+            name="table2",
+            location="s3://some_location/table2",
+            upgraded_to="ucx_default.db1_dst.dst_table2",
+        ),
+        Table(
+            object_type="EXTERNAL",
+            table_format="DELTA",
+            catalog="hive_metastore",
+            database="schema1",
+            name="table3",
+            location="s3://some_location/table3",
+        ),
+    ]
+    custom_migration_status_refresher = create_autospec(MigrationStatusRefresher)
+    migration_index = MigrationIndex(
+        [
+            MigrationStatus("schema1", "table1", "ucx_default", "db1_dst", "dst_table1"),
+            MigrationStatus("schema1", "table2", "ucx_default", "db1_dst", "dst_table2"),
+        ]
+    )
+    custom_migration_status_refresher.index.return_value = migration_index
+
+    # Call the method
+    ctx = run_workflow(
+        TableMigration.refresh_migration_status,
+        table_crawler=custom_table_crawler,
+        migration_status_refresher=custom_migration_status_refresher,
+    )
+
+    # # Assert the log message
+    # log_messages = [record.getMessage() for record in log_capture_handler.records]
+    # assert 'remained-table-to-migrate: schema1.table3' in log_messages
+    return ctx.task_run_warning_recorder.snapshot()
+
```

Comment:
I am trying to get the logs here and assert from the ctx. I came across `task_run_warning_recorder ` and thought might help but it requires config flag... ` ValueError: config flag is required`

Reply:
we use the [Pytest caplog fixture](https://docs.pytest.org/en/6.2.x/logging.html#caplog-fixture) to assert log messages, search for "caplog" in the tests folder:

`` python
with caplog.at_level(logging.WARNING, logger="databricks.labs.ucx.<module to test>"):
    # Run code that creates log messages
    
assert "this (partial) message should be in the logs" in caplog.text
# OR
assert "this (complete) message should be in the logs" in caplog.messages
 ``

Reply:
Done and moved under `test_table_migrate.py`

Reply:
Done - under `test_table_migrate.py`